### ISSUE TITLE
Mirador not working in Bootstrap5 Tabs

### ISSUE THREAD

**Issue by michelaparma** [ProjectMirador/mirador#3540]
I would like to organise the content page through the Bootstrap5 Tabs and insert the Mirador viewer in the third tab. 
In this way the viewer does not work and creates _TypeError: Cannot read properties of null (reading 'resetAfterIndex')_. 
But if the viewer is in the first tab, the one that is open and displayed when the page is loaded, then it works perfectly. 
Is there a solution to make the viewer work even if it is not displayed immediately?

Problem:
![mirador-tab-problem](https://user-images.githubusercontent.com/44635720/155524842-65e876e0-eea2-418e-8177-79a0493fb5c5.png)

Error:
![typeError](https://user-images.githubusercontent.com/44635720/155524067-966c5681-e63a-4a46-8683-e639a9c3ceb8.png)

**Comment by marlo-longley** [comment]
Related to #3506 (both tickets relate to putting mirador into an initially hidden DOM element).  Components relying on measurement (OpenSeadragon for viewport, react-window for virtualization) can't initialize properly.

Possible longer term solution (Claude)
Add IntersectionObserver to ThumbnailNavigation that automatically detects visibility and recalculates dimensions. This would make it "just work" without requiring users to call anything. The existing ResizeObserver in WorkspaceElastic shows the pattern is already used elsewhere in the codebase.

Simpler fix:
Could add a public processResize() method to MiradorViewer that users can call after unhiding similar to the idea in #3506 

Implementer side fix:
Use `visibility: hidden` instead of `display: none` (allows dimensions) -- but not sure what Bootstrap does

**Comment by dnoneill** [comment]
There is a solution to this with bootstrap tabs but I wonder if we should be doing something similar in Mirador

```
const viewerTab = document.getElementById('viewer-tab');
    let miradorInstance = null;

  viewerTab.addEventListener('shown.bs.tab', function () {
    Mirador.viewer(config);
    if (!miradorInstance) {
      // First time the tab is shown: create the viewer
      miradorInstance = Mirador.viewer({
        id: 'mirador-viewer',
        windows: [
          {
            loadedManifest: 'https://iiif.harvardartmuseums.org/manifests/object/299843'
          }
        ]
      });
    }
  });
```

**Comment by marlo-longley** [comment]
Thanks Niqui, I think it would be good to understand what the options are on Mirador's end to have this "just work" for users. Might be some ideas in earlier comments and linked tickets. If nothing seems like a good fit we can at least document this solution.

**Comment by dnoneill** [comment]
Yeah I have been looking into it and I believe I have a solution.
